webknossos/webknossos/dataset/dataset.py

```diff
@@ -1002,3 +1118,203 @@ def _initialize_layer_from_properties(self, properties: LayerProperties) -> Laye
             raise RuntimeError(
                 f"Failed to initialize layer: the specified category ({properties.category}) does not exist."
             )
+
+
+class RemoteDataset(Dataset):
+    """Representation of a dataset on the webknossos server, returned from `Dataset.remote_open()`.
+    Read-only image data is streamed from the webknossos server using the same interface as `Dataset`.
+    Additionally, metadata can be set via the additional properties below."""
+
+    def __init__(
+        self,
+        dataset_path: UPath,
+        dataset_name: str,
+        organization_id: str,
+        sharing_token: Optional[str],
+        context: ContextManager,
+    ) -> None:
+        """Do not call manually, please use `Dataset.remote_open()` instead."""
+        try:
+            super().__init__(
+                dataset_path,
+                voxel_size=_UNSPECIFIED_SCALE_FROM_OPEN,
+                exist_ok=True,
+                read_only=True,
+            )
+        except FileNotFoundError:
+            warnings.warn(
+                f"Cannot open remote webknossos dataset {dataset_path} as zarr. "
+                + "Returning a stub dataset instead, accessing metadata properties might still work.",
+                RuntimeWarning,
+            )
+            self.path = None  # type: ignore[assignment]
+        self._dataset_name = dataset_name
+        self._organization_id = organization_id
+        self._sharing_token = sharing_token
+        self._context = context
```

Comment:
Is the ContextManager serializable?

Reply:
Do you mean with pickle? Yes, just checked it. However, using a remote dataset with a webknossos-context (e.g. a different wk-url as in the normal context) didn't work before, since the context-manager wasn't [reentrant](https://docs.python.org/3/library/contextlib.html#reentrant-context-managers). I fixed this in the latest commit